components/camel-micrometer/src/test/java/org/apache/camel/component/micrometer/routepolicy/AbstractMicrometerRoutePolicyTest.java

```diff
@@ -57,10 +52,25 @@ protected String formatMetricName(String name) {
 
     @Override
     protected CamelContext createCamelContext() throws Exception {
+        meterRegistry = new SimpleMeterRegistry();
         CamelContext context = super.createCamelContext();
         MicrometerRoutePolicyFactory factory = createMicrometerRoutePolicyFactory();
+        factory.setCamelContext(context);
```

Comment:
Hey guys, 
Just want to let you know that when I updated from Camel 4.4 to 4.8 and the `MicrometerRoutePolicy` started failing to initialize because previously the `MicrometerRoutePolicyFactory` didn't require `camelContext` to be provided.

Considering that `MicrometerRoutePolicyFactory` implements `CamelContextAware` and extends `ServiceSupport`, maybe it worth to do automatic wiring when the factory gets added to the Camel Context via `addRoutePolicyFactory`, or some other way.


PS. Of course I'll fix it my code by providing the `camelContext` to it.